source/Company.py

```diff
@@ -1,53 +1,184 @@
+        '''
+        if stock_price > self.net_worth() or check_positive(stock_price) is False:
+            return False
+        else:
+            net_worth = self.net_worth()
+            self.stock_price = stock_price
+            self.stocks_num = int(net_worth//self.stock_price)
+            return True
 
     def set_comp_type(self, comp_type):
-
-        pass
+        '''
+        Updates the type of the copmany, as long as its valid
+        '''
+        if not check_string(comp_type):
+            return False
+        else:
+            self.comp_type = comp_type
+            return True
 
     def update_net_worth(self, net_worth):
-
-        pass
+        '''
+        Updates the networth of the company
+        '''
+        if not check_positive(net_worth):
+            return False
+        else:
+            self.stock_price = net_worth / self.stocks_num
+            return True
 
     def add_stocks(self, number):
-
-        pass
+        '''
+        The function get int (numeber) and add it
+        to the current number of stocks
+        '''
+        if check_int(number) is False:
+            return False
+        new_stock_num = self.stocks_num + number
+        if new_stock_num <= 0:
+            return False
+        else:
+            self.stocks_num = new_stock_num
+            return True
+
+    def __str__(self):
+        '''
+        A string with all the required data of the company
+        '''
+        return f"{self.name}, {self.stocks_num} stocks, Price: {self.stock_price}, {self.comp_type}, Net Worth: {self.net_worth()}"
 
     def __repr__(self):
-
-        pass
+        '''
+        A string with all the required data of the company
+        '''
+        return f"{self.name}, {self.stocks_num} stocks, Price: {self.stock_price}, {self.comp_type}, Net Worth: {self.net_worth()}"
 
     def __lt__(self, other):
-
-        pass
+        '''
+        Operator overloading - less than
+        '''
+        if self.net_worth() < other.net_worth():
+            return True
+        return False
 
     def __gt__(self, other):
-
-        pass
+        '''
+        Operator overloading - greater than
+        '''
+        if self.net_worth() > other.net_worth():
+            return True
+        return False
 
     def __eq__(self, other):
-
-        pass
+        '''
+        Operator overloading - equal to
+        '''
+        if self.net_worth() == other.net_worth():
+            return True
+        return False
 
     def __add__(self, other):
-        pass
+        '''
+        Operator overloading which merges two companies into one
+        '''
+        new_company = Company(name=self.name,
+                              stocks_num=self.stocks_num + other.stocks_num,
+                              stock_price=(self.net_worth() + other.net_worth()) /
+                              (self.stocks_num + other.stocks_num),
+                              comp_type=self.comp_type)
```

Comment:
This section was super confusing for me to read.
You started out good by going down a line after each coma ( , ) BUT!, on line 182, where I though you are giving the function another argument it happend to be a long line which you broke down into two.
In this kind of cases add another tab:
```python
new_company = Company(name=self.name,
                     stocks_num=self.stocks_num + other.stocks_num,
                     stock_price=(self.net_worth() + other.net_worth()) /
                         self.stocks_num + other.stocks_num,
                     comp_type=self.comp_type)
```

Reply:
If it will make the code clearer in the cost of an extra line. than be it.

Reply:
fixed